### ISSUE TITLE
Gastronomy: Should be compatible with Replimat

### ISSUE THREAD

**Issue by OrionFive** [OrionFive/Hospitality#486]
Currently breaks serving.
At the very least it should break anything. Better would be if it somehow worked. Waiters should pull food from the Replimat if it's not in stock.

**Comment by sumghai** [comment]
Hi @OrionFive , Replimat author here - apologies, I was meant to actually try Gastronomy myself before writing up a full bug report, but I've been knee-deep with a couple of side projects. Here are my suggestions:

* Replimats are supposed to be **self-service** food terminals (as per their depiction in _Star Trek: Deep Space Nine_), so visitors should just go up and use the terminals themselves. **Waiters should not be involved in any way.**

* Gastronomy's restaurant menu system should (as usual) restrict what meals visitors can pull from terminals.

* Ideally, visitors automatically pay when they use the terminals, and skip the cash register altogther.

For your reference, here's relevant code for the Replimat terminals and Utils/Toils/JobDriver Harmony patches:

https://github.com/sumghai/Replimat/blob/master/Source/Building_ReplimatTerminal.cs

https://github.com/sumghai/Replimat/blob/master/Source/HarmonyPatches.cs

What changes do you think are needed on Replimat's end?

**Comment by OrionFive** [comment]
Hey @sumghai, thanks for chiming in and sharing your thoughts. 
From what I can tell, all work is on my end. But your vision means new jobs for the guests, which is a lot of work to make and also to maintain.
Even just hooking up the replimat with the menu and stockpile is probably going to be quite annoying to do.
So when I get around to this I'll stick with "not breaking people's games". I don't have time for more than that, especially since it costs extra maintenance time in the future as well.

**Comment by sumghai** [comment]
Fair enough, @OrionFive. I definitely feel your pain when it comes to feature creep and maintainability :)

**Comment by OrionFive** [comment]
Hospitality is now compatible with the Replimat. Guests can purchase food from it directly. However, it's not integrated with Gastronomy yet.
